theano/sandbox/cuda/opt.py

```diff
@@ -97,6 +97,7 @@
 register_opt(name='gpu_constant_folding')(
     tensor.opt.constant_folding)
 register_opt()(theano.tensor.opt.local_subtensor_make_vector)
+register_opt('unsafe')(theano.tensor.opt.local_remove_all_assert)
```

Comment:
You should not call register_opt as this enable this opt by default. You should do sometime similar to what register_opt do. In this case:
```
gpu_optimizer.register('local_remove_all_assert', theano.tensor.opt.local_remove_all_assert, 'unsafe',
                               'gpu', *tags)
```

similar for the other gpuarray folder. And add a test that by default, this optimization isn't done.